src/docker.ts

```diff
@@ -26,13 +32,30 @@ const saveDockerImages = async (): Promise<void> => {
       `Cache miss occurred on the primary key ${key}. Not saving cache as read-only option was selected.`
     );
   } else {
-    await execBashCommand(
-      'docker image list --format "{{ .Repository }}:{{ .Tag }}" | ' +
-        '2>&1 xargs --delimiter="\n" --no-run-if-empty --verbose --exit ' +
-        `docker save --output ${DOCKER_IMAGES_PATH}`
+    const preexistingImages = getState(DOCKER_IMAGES_LIST).split("\n");
+    const images = await execBashCommand(
+      'docker image list --format "{{ .Repository }}:{{ .Tag }}"'
     );
```

Comment:
The logs read clearly to those who are acquainted with the internals of docker-cache, but look at the corresponding log output of the Save Cache job, and I think you will see they are somewhat unwelcoming to newcomers. It would help to offer some context on why we are executing this command both here and in `loadDockerImages`.